nat-lab/docker-compose.yml

```diff
@@ -11,168 +11,94 @@ services:
-        # therefore it takes the specified IP address and an
-        # IP address + 1. The logic for this is inside bin/stun-server.
-        ipv4_address: 10.0.1.1
-    logging:
-      options:
-        max-size: 50m
-
-  # Run two DERP servers configured to run in a cluster
-  derp-01:
-    hostname: derp-01
-    image: nat-lab:base
-    entrypoint: /opt/bin/derp-server
-    networks:
-      internet:
-        ipv4_address: 10.0.10.1
-    volumes:
-      - ./data/nordderper/config1.yml:/etc/nordderper/config.yml
-    working_dir: /etc/nordderper
-  derp-02:
-    hostname: derp-02
-    image: nat-lab:base
-    entrypoint: /opt/bin/derp-server
-    networks:
-      internet:
-        ipv4_address: 10.0.10.2
-    volumes:
-      - ./data/nordderper/config2.yml:/etc/nordderper/config.yml
-    working_dir: /etc/nordderper
-  derp-03:
-    hostname: derp-03
-    image: nat-lab:base
-    entrypoint: /opt/bin/derp-server
-    networks:
-      internet:
-        ipv4_address: 10.0.10.3
-    volumes:
-      - ./data/nordderper/config3.yml:/etc/nordderper/config.yml
-    working_dir: /etc/nordderper
-
-  # Two VPN servers
-  vpn-01:
-    hostname: vpn-01
+  ###########################################################
+  #                         CLIENTS                         #
+  ###########################################################
+  # Clients for cone network, they'll use cone gateways
+  # as their route to the open internet
+  cone-client-01: &common-client
+    hostname: cone-client-01
     image: nat-lab:base
-    entrypoint: /opt/bin/vpn-server
+    entrypoint: /opt/bin/client
+    devices:
+      - "/dev/net/tun"
+    cap_drop:
+      - ALL
+    cap_add:
+      - NET_RAW
+      - NET_ADMIN
+      - NET_BIND_SERVICE
+      - SETUID
+      - SETGID
+    sysctls:
+      # Disable return path filter. It blocks inbound traffic when using fwmark routing policy.
+      # net.ipv4.conf.all.rp_filter.src_valid_mark=1 is supposed to fix this, but it doesn't
+      # seem to work. So for now disable return path filter outright.
+      - net.ipv4.conf.all.rp_filter=0
+      - net.ipv4.conf.eth0.rp_filter=0
+      # Enable IPv6, by default, docker has this set to '1'
+      - net.ipv6.conf.all.disable_ipv6=0
+      - net.ipv6.conf.default.disable_ipv6=0
+      - net.ipv6.conf.all.forwarding=1
+    security_opt:
+      - no-new-privileges:true
     networks:
-      internet:
-        ipv4_address: 10.0.100.1
+      cone-net-01:
+        ipv4_address: 192.168.101.104
     environment:
-      CLIENT_GATEWAY_PRIMARY: none
-    privileged: true
+      CLIENT_GATEWAY_PRIMARY: 192.168.101.254
+    extra_hosts:
+      - "photo-album:10.0.80.80"
+      - "stun-01:10.0.1.1"
+      - "derp-01:10.0.10.1"
+      - "derp-02:10.0.10.2"
+      - "derp-03:10.0.10.3"
+      - "derp-00:10.0.10.245"
     volumes:
       - ../:/libtelio
-  vpn-02:
-    hostname: vpn-02
-    image: nat-lab:base
-    entrypoint: /opt/bin/vpn-server
+  cone-client-02:
```

Comment:
Why doesn't this client have the same cap_*, devices, security and sysctls sections as the `cone-client-01`?

Reply:
It has, it inherits its values through the `common-client` anchor.

Reply:
Oops, right. I didn't understand anchors...

Reply:
Oops, right. I didn't understand anchors...